src/main/java/org/opentripplanner/transit/service/TransitModel.java

```diff
@@ -0,0 +1,564 @@
+import org.opentripplanner.routing.trippattern.Deduplicator;
+import org.opentripplanner.routing.util.ConcurrentPublished;
+import org.opentripplanner.routing.vertextype.TransitStopVertex;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.model.framework.TransitEntity;
+import org.opentripplanner.transit.model.network.TransitMode;
+import org.opentripplanner.transit.model.organization.Agency;
+import org.opentripplanner.transit.model.organization.Operator;
+import org.opentripplanner.transit.model.site.Stop;
+import org.opentripplanner.transit.model.site.StopLocation;
+import org.opentripplanner.updater.GraphUpdaterConfigurator;
+import org.opentripplanner.updater.GraphUpdaterManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Repository for Transit entities.
+ */
+public class TransitModel implements Serializable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TransitModel.class);
+
+  private static final long serialVersionUID = 1L;
+
+  private final Collection<Agency> agencies = new ArrayList<>();
+  private final Collection<Operator> operators = new ArrayList<>();
+  private final Collection<String> feedIds = new HashSet<>();
+  private final Map<String, FeedInfo> feedInfoForId = new HashMap<>();
+
+  /**
+   * Allows a notice element to be attached to an object in the OTP model by its id and then
+   * retrieved by the API when navigating from that object. The map key is entity id:
+   * {@link TransitEntity#getId()}. The notice is part of the static transit data.
+   */
+  private final Multimap<TransitEntity, Notice> noticesByElement = HashMultimap.create();
+  private final Map<Class<?>, Serializable> services = new HashMap<>();
+  private final TransferService transferService = new TransferService();
+
+  /** List of transit modes that are availible in GTFS data used in this graph **/
+  private final HashSet<TransitMode> transitModes = new HashSet<>();
+
+  /**
+   * Map from GTFS ServiceIds to integers close to 0. Allows using BitSets instead of
+   * {@code Set<Object>}. An empty Map is created before the Graph is built to allow registering IDs
+   * from multiple feeds.
+   */
+  private final Map<FeedScopedId, Integer> serviceCodes = Maps.newHashMap();
+
+  /** Pre-generated transfers between all stops. */
+  public final Multimap<StopLocation, PathTransfer> transfersByStop = HashMultimap.create();
+
+  private StopModel stopModel;
+  // transit feed validity information in seconds since epoch
+  private long transitServiceStarts = Long.MAX_VALUE;
+  private long transitServiceEnds = 0;
+
+  /** Data model for Raptor routing, with realtime updates applied (if any). */
+  private final transient ConcurrentPublished<TransitLayer> realtimeTransitLayer = new ConcurrentPublished<>();
+
+  public final transient Deduplicator deduplicator;
+  private transient CalendarService calendarService;
+
+  public transient TransitModelIndex index;
```

Comment:
I know that this was like that before, but should this really be public and not encapsulated a bit better?

Reply:
I think we can encapsulate this better when we have completed the refactoring. Then we can force all access through the `transitService` where needed.

Reply:
Indeed encapsulation of the *Model classes will be performed in a subsequent refactoring round